prism/src/main/java/org/apache/falcon/service/FeedSLAAlertService.java

```diff
@@ -0,0 +1,224 @@
+import org.apache.falcon.entity.FeedInstanceStatus;
+import org.apache.falcon.entity.store.ConfigurationStore;
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.Frequency;
+import org.apache.falcon.entity.v0.feed.Cluster;
+import org.apache.falcon.entity.v0.feed.Feed;
+import org.apache.falcon.entity.v0.feed.LocationType;
+import org.apache.falcon.expression.ExpressionHelper;
+import org.apache.falcon.jdbc.MonitoringJdbcStateStore;
+import org.apache.falcon.persistence.FeedSLAAlertBean;
+import org.apache.falcon.persistence.PendingInstanceBean;
+import org.apache.falcon.util.Pair;
+import org.apache.falcon.util.ReflectionUtils;
+import org.apache.falcon.util.StartupProperties;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+  * Service to know which all feeds have missed SLA.
+  */
+public final class FeedSLAAlertService implements FalconService, FeedSLAAlert {
+
+    private static final String NAME = "FeedSLAAlertService";
+
+    private static final Logger LOG = LoggerFactory.getLogger(FeedSLAAlertService.class);
+
+    private Multimap<Pair, Date> feedInstancesToBeMonitored;
+
+    private MonitoringJdbcStateStore store = new MonitoringJdbcStateStore();
+
+    private Set<FeedSLAAlert> listeners = new LinkedHashSet<FeedSLAAlert>();
+
+/**
+  * Permissions for storePath.
+  */
+    private static final FsPermission STORE_PERMISSION = new FsPermission(FsAction.ALL, FsAction.NONE, FsAction.NONE);
+
+
+    private static final FeedSLAAlertService SERVICE = new FeedSLAAlertService();
+
+    public static FeedSLAAlertService get() {
+        return SERVICE;
+    }
+
+    private EntityManager getEntityManager() {
+        return FalconJPAService.get().getEntityManager();
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public void init() throws FalconException {
+        String listenerClassNames = StartupProperties.get().
+                getProperty("feedAlert.listeners");
+        for (String listenerClassName : listenerClassNames.split(",")) {
+            listenerClassName = listenerClassName.trim();
+            if (listenerClassName.isEmpty()) {
+                continue;
+            }
+            FeedSLAAlert listener = ReflectionUtils.getInstanceByClassName(listenerClassName);
+            registerListener(listener);
+        }
+        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+        executor.scheduleWithFixedDelay(new Monitor(), 0, 20, TimeUnit.SECONDS);
+    }
+
+    public void registerListener(FeedSLAAlert listener) {
+        listeners.add(listener);
+    }
+
+    @Override
+    public void destroy() throws FalconException {
+
+    }
+
+
+    private class Monitor implements Runnable {
+
+        @Override
+        public void run() {
+            processSLALowCandidates();
```

Comment:
These 2 methods have some overlapping logic with methods getFeedSLAMissPendingAlerts and getSLAStatus of FeedSLAMonitoringService. Please reuse those methods, so that the API and this service are in sync.